### PR TITLE
Create a proper logging system for de-stats

### PR DESCRIPTION
**Pull request by jack-mitt**
The logger lies in /logs/main.go I am passing it around as a global variable accessed using logs.Logger. Debug mode can be enabled using --debug=on in the command line arguments. There may not be enough debug statements since this was added after writing the main code.

### CODE REVIEW

File: api/main.go [hunk 1]
```diff
@@ -31,14 +18,6 @@ type ErrorResponse struct {
...
 // RootHandler handles get requests to the / endpoint.
 func RootHandler(ctx echo.Context) error {
-
-	// Extract the days query parameter.
-	days, err := util.IntQueryParam(ctx, "days", 1, 0, 365)
-	fmt.Println(days)
-	if err != nil {
-		return ctx.JSON(http.StatusBadRequest, ErrorResponse{Description: err.Error()})
-	}
-
```
Comment (ianmcorvidae) [inline api/main.go:41]: These changes (both in this file) look unrelated -- what's going on there?

**Review by ianmcorvidae — COMMENTED** [review]
Looks pretty reasonable to me. One question in there. I also recall there being some Println and such scattered throughout the code, and I'm not sure if those have been replaced. If not, they should be.

File: api/main.go (hunk 1 above)
Comment (jack-mitt) [reply api/main.go:41]: Those were for a testing parameter I was using to get it working on the root response. Just thought I should remove them now. And yes the println statements have been replaced by log statements

File: api/main.go [hunk 2]
```diff
@@ -31,14 +18,6 @@ type ErrorResponse struct {
 
 // RootHandler handles get requests to the / endpoint.
 func RootHandler(ctx echo.Context) error {
```
Comment (slr71) [inline api/main.go:20]: This doesn't have to be done as part of this pull request, but I'd like to see the root endpoint return something a little more useful. The permissions service returns a little bit of information about the service that it extracts from the Swagger spec. One option would be to do something similar here, and read the service information in from the `swagger.json` file. You could do this once upon startup and return the same object every time. The implementation in the permissions service is here: https://github.com/cyverse-de/permissions/blob/master/restapi/impl/status/get_status.go. The only difference here is that you'd have to read the JSON in from the file.

**Review by slr71 — APPROVED** [review]
I really like what you've got here. The code is organized well and easy to read, which is hugely important. I had one minor suggestion for the `/` endpoint, which is to have it return some basic information about the service itself, which may help with diagnostics as changes are made in the future. That suggestion isn't really related to this PR, however, and it doesn't have to be done immediately. Once that change is made, I think we can call this project complete! Great work on this!
